### PR TITLE
Introduce clang format

### PR DESCRIPTION
**Pull request by zigen**
# TL;DR
* Introduce [clang-format](https://clang.llvm.org/docs/ClangFormat.html) to format our code automatically.
* Introduce [editorconfig](https://editorconfig.org/) to share editor configuration
* check C++ source code formnat on CI

# motivation
the current code base has a variety of code style. For example, sometimes it used 4 spaces for indentation, but sometimes it is 8.
this situation increases reviewing cost and decreases the readability of codes.
I plan to contribute a more substantial part of this project. Before doing that, I desire to introduce clang-format or similar tools.

# discussion
now I choose [google's c++ style](https://google.github.io/styleguide/cppguide.html) to format because current codebase seems similar to it.
if there's any request for coding style (`.clang-format` file), let me know.

# misc
* we can integrate clang-format into OMNET++IDE with [CppStyle](https://github.com/wangzw/CppStyle). 
* I want to write how to format code for someone who wants to contribute. which file is suitable? (maybe this related to #76).
* We also need [clang-tidy](https://clang.llvm.org/extra/clang-tidy/) to check naming conventions because the current code has tons of inconsistent names. For example, `BSAresult`, `return_QNIC_address_to_destAddr`. these should be `BsaResult`, `getQNicAddressToDestAddr`. please don't mix CamelCase and lower_case. this naming confuses contributors. I want to introduce clang-tidy in other PR.

### CODE REVIEW

**Comment by zigen** [comment]
now I only commit the clang-format configuration, so CI should fail.
after confirming CI fails, I'll push the formatted code.

File: Makefile [hunk 1]
```diff
@@ -1,158 +1,7 @@
-#
```
Comment (zigen) [inline Makefile:1]: I think this Makefile is no longer used and we use `quisp/Makefile`.
so I remove this file and put new Makefile for clang-format.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  SplitEmptyNamespace: true
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Attach
+BreakBeforeInheritanceComma: false
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: BeforeColon
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     180
```
Comment (zigen) [inline .clang-format:47]: the code base has many long comments, so I extend the value to 180 from 80.

**Comment by zigen** [comment]
hmm, now CI broken unexpectedly... later I'll fix it.

File: quisp/modules/ConnectionManager.cc [hunk 3]
```diff
@@ -742,62 +726,67 @@ void ConnectionManager::intermediate_reject_req_handler(RejectConnectionSetupReq
...
+  RuleSet *tomography = new RuleSet(RuleSet_id, owner, partner);
+  Rule *Random_measure_tomography = new Rule(RuleSet_id, rule_index);
 
-  Condition* measurement_condition = new Condition();//Technically, there is no condition because an available resource is guaranteed whenever the rule is ran.
-  Clause* count_clause = new MeasureCountClause(num_of_measure, partner, qnic_type , qnic_index, 0);//3000 measurements in total. There are 3*3 = 9 patterns of measurements. So each combination must perform 3000/9 measurements.
-  Clause* resource_clause = new EnoughResourceClause(partner, num_resources);
+  Condition *measurement_condition = new Condition();  // Technically, there is no condition because an available resource is guaranteed whenever the rule is ran.
+  Clause *count_clause =
+      new MeasureCountClause(num_of_measure, partner, qnic_type, qnic_index,
+                             0);  // 3000 measurements in total. There are 3*3 = 9 patterns of measurements. So each combination must perform 3000/9 measurements.
```
Comment (zigen) [inline quisp/modules/ConnectionManager.cc:761]: These lines are formatted uglily because the comment is put on the same line.
I think clang-format cannot fix this properly, so I'll fix it manually later.

File: quisp/rules/Action.cc [hunk 4]
```diff
@@ -7,811 +7,860 @@
...
+  bool meas_X, meas_Z = false;
+
+  meas_X = trash_qubit_X->Xpurify(qubit);  // Error propagation only. Not based on density matrix
+  meas_Z = trash_qubit_Z->Zpurify(qubit);  // Error propagation only. Not based on density matrix
+
+  qubit->Lock(ruleset_id, rule_id, action_index);
+
+  // Trash qubit has been measured. Now, break the entanglement info of the partner.
+  // There is no need to overwrite its density matrix since we are only tracking errors.
+  if (trash_qubit_Z->entangled_partner != nullptr) {
```
Comment (zigen) [inline quisp/rules/Action.cc:261]: there're tons of copy-and-paste codes. 
I'll refactor these codes in other PR.

**Comment by Chibikuri** [comment]
Thank you for formatting. I agree with using Google's c++ style. Let me review the code changes later. 
I created [CONTRIBUTE](https://github.com/sfc-aqua/quisp/blob/master/.github/CONTRIBUTING.md). 
This is a pre-version of it, but it would be great if you mention some notes about Google's  c++ style.

**Review by rdviii — APPROVED** [review]
